### ISSUE TITLE
Invalid validations on KeyPoints class?

### ISSUE THREAD

**Issue by Chappie74** [roboflow/supervision#1440]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar bug report.

### Bug

I'm trying to use the KeyPoints class, with my xy data as [ [x1,y1], [x2,y1], ....  ] 
Based on the information in the docs string this seems to be input needed. However, I run into an error.
```
ValueError: xy must be a 2D np.ndarray with shape ((68, 2),), but got shape (68, 2)
```

![image](https://github.com/user-attachments/assets/6a7c65e1-78c0-4fdc-8c04-e152a20a3f3d)

The validation for the shape check does not seems to be consistent. Either that or I may be doing something wrong
![image](https://github.com/user-attachments/assets/6e3f2b97-9510-4203-ae2c-feb0c4c9c5ef)

### Environment

Supervision: 0.22.0
Python: 3.10

### Minimal Reproducible Example

```python
sv.KeyPoints(xy=np.array([[1,3], [1,3], [1,3]]))
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by LinasKo** [comment]
Indeed, the docs are not correct. KeyPoints objects expects to hold N skeletons, with dimensions of `(number_of_skeletons x number_of_keypoints x keypoint_dimensions)`.

**Comment by LinasKo** [comment]
It's definitely possible, though it works best if you have a model already, and can call one of our [connector methods](https://supervision.roboflow.com/latest/keypoint/core/): `from_inference`, `from_ultralytics`, `from_mediapipe`, etc.

I can't recall building a `sv.KeyPoints` by hand, but it shouldn't be too hard.

```
import supervision as sv
import numpy as np
import cv2

# wget https://media.roboflow.com/notebooks/examples/dog.jpeg
image = cv2.imread("dog.jpeg")

points = np.array([
    # Object 1
    [[100, 100], [200, 100], [200, 200], [100, 200]],

    # Object 2
    [[150, 150], [250, 150], [250, 250], [150, 250]]
])
confidence = np.array([
    [0.9, 0.8, 0.7, 0.6],
    [0.9, 0.8, 0.7, 0.6]
])
class_id = np.array([0, 1])


key_points = sv.KeyPoints(
    xy=points,
    confidence=confidence,
    class_id=class_id
)

edge_annotator = sv.EdgeAnnotator(
    edges=[
        (0, 1),
        (1, 2),
        (2, 3),
        (3, 0)
    ]
)
image = edge_annotator.annotate(image.copy(), key_points)

sv.plot_image(image)
```

**Comment by Chappie74** [comment]
Makes sense now. Thank you.

**Comment by LinasKo** [comment]
@SkalskiP, This is very small, so I'm taking it and deploying together with other small fixes.
Addressed by: #1448
